### PR TITLE
[JUJU-3512] validate secret backend config when we change the secret backend in model config

### PR DESCRIPTION
**Pull request by ycliuhw**
This PR validates the secret backend connection information when we want to set it as a model's default secret backend.

## Checklist

- [x] Code style: imports ordered, good names, simple structure, etc
- [x] Comments saying why design decisions were made
- [x] Go unit tests, with comments saying what you're testing
- [ ] ~[Integration tests](https://github.com/juju/juju/tree/develop/tests), with comments saying what you're testing~
- [ ] ~[doc.go](https://discourse.charmhub.io/t/readme-in-packages/451) added or updated in changed packages~

## QA steps

```sh
juju model-config secret-backend=myvault
ERROR cannot ping backend "myvault": backend not reachable: Get "http://10.180.97.1:8200/v1/sys/health?drsecondarycode=299&performancestandbycode=299&sealedcode=299&standbycode=299&uninitcode=299": dial tcp 10.180.97.1:8200: connect: connection refused
```

## Documentation changes

No

## Bug reference

https://bugs.launchpad.net/juju/+bug/2015965

### CODE REVIEW

**Comment by ycliuhw** [comment]
/build

12:59:38 FAIL: worker_test.go:442: WorkerSuite.TestAddMachine

File: apiserver/facades/client/modelconfig/modelconfig.go [hunk 1]
```diff
@@ -345,15 +346,24 @@ func (c *ModelConfigAPI) checkSecretBackend() state.ValidateConfigFunc {
 		}
 		backendName, ok := v.(string)
 		if !ok {
-			return errors.NotValidf("%q is not a string", config.SecretBackendKey)
+			return errors.NotValidf("%q config value is not a string", config.SecretBackendKey)
```
Comment (wallyworld) [inline apiserver/facades/client/modelconfig/modelconfig.go:349]: I think we need `NewNotValidf` here

**Review by wallyworld — APPROVED** [review]
Thanks for fixing

**Pull request merged**